### PR TITLE
Fix scoped arena iterator

### PR DESCRIPTION
**Pull request by yuslepukhin**
This brings down write_batch_test immediately.
Due to the fact that memory does not go away the code continues to operate on a destroyed internal iterator.

### CODE REVIEW

**Comment by yuslepukhin** [comment]
@siying @IslamAbdelRahman If you do not mind, I will merge it

File: table/scoped_arena_iterator.h [hunk 1]
```diff
@@ -8,20 +8,46 @@
 #pragma once
 
 #include "table/internal_iterator.h"
+#include "port/port.h"
 
 namespace rocksdb {
 class ScopedArenaIterator {
+
+  void reset(InternalIterator* iter) ROCKSDB_NOEXCEPT {
```
Comment (IslamAbdelRahman) [inline table/scoped_arena_iterator.h]: Can we move this down to the private section ?

**Comment by IslamAbdelRahman** [comment]
Thanks @yuslepukhin, LGTM. I have a minor comment

**Comment by siying** [comment]
Oops, I missed @IslamAbdelRahman 's comment and merged it. I guess we should address it in a follow-up pull request.

**Pull request merged**